src/QuickMarcEditor/utils.js

```diff
@@ -1092,15 +1096,16 @@ export const groupSubfields = (field, authorityControlledSubfields = []) => {
   });
 };
 
-export const splitFields = marcRecord => {
+export const splitFields = (marcRecord, linkingRules) => {
   return {
     ...marcRecord,
     records: marcRecord.records.map(record => {
       if (!record._isLinked) {
         return record;
       }
 
-      const subfieldGroups = groupSubfields(record, record.authorityControlledSubfields);
+      const linkingRule = linkingRules.find(rule => rule.id === record.linkingRuleId);
+      const subfieldGroups = groupSubfields(record, linkingRule?.authoritySubfields);
```

Comment:
We also need to take into account subfield transformations
There's an example in `useAuthorityLinking` where we use `getControlledSubfields` function to apply subfield transformations to get actual controlled subfields
Let's move that function to utils and re-use it

We can test this case by linking bib 240 to auth 100 with $t, then adding a `$a` in bib field and saving. It should show a toast error that controlled subfield can't be edited

And same comment applies to function `validateSubfieldsThatCanBeControlled`

Reply:
Good catch, fixed